examples/rosettacode/rock_paper_scissor.nit

```diff
@@ -0,0 +1,56 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+# This program is public domain
+#
+# Task: Programming_Task/Rock-paper-scissors
+# SEE: <http://rosettacode.org/wiki/Rock-paper-scissors>
+#
+# author: Istvan szalai 22/05/2015
+
+# Return a value to dertermine who win the match.
+# Return 0 for lose, 1 for win, 2 for equality.
+fun win_or_lose(player, computer: String): Int do
+	if player == computer then return 2
+	if player == "rock" and computer == "scissors" then return 1
+	if player == "rock" and computer == "paper" then return 0
+	if player == "paper" and computer == "rock" then return 1
+	if player == "paper" and computer == "scissors" then return 0
+	if player == "scissors" and computer == "paper" then return 1
+	return 0
+end
+
+# Return a string with about the match winner or looser.
+fun result_of_the_match_message(number: Int): String do
+	if number == 2 then return "It's a tie!"
+	if number == 1 then return "You win!"
+	return "You lose!"
+end
+
+# Return a String for the player about what the computer player did.
+fun computer_message(number: String): String do
+	if number == "rock" then return "The computer played rock"
+	if number == "scissors" then return "The computer played scissors"
+	return "The computer played paper"
+end
+
+# Return a random value for the computer player input.
+fun computer_player : String do
+	var tab = ["rock", "scissors", "paper"]
+	return tab.rand
+end
+
+# Game loop.
+var x = ""
+var test = false
+loop
+	while not test do
+		print "Choose your weapon|exit"
+		x = gets
+		if x == "exit" then break
+		if x == "paper" or x == "rock" or x == "scissors" then test = true
```

Comment:
I'm a bit surprised to hear that ! I thought that the array was
instantiated once only, but I guess it makes sense since arrays are indeed
mutable.

Maybe we should proceed differently then ? Have the literal arrays
immutable ? If we go with a literal array, most of the time it is for
read-only purposes, in this sense it could be varonced ?

2015-06-03 10:49 GMT-04:00 Jean Privat <notifications@github.com>:

> In examples/rosettacode/rock_paper_scissor.nit
> <https://github.com/privat/nit/pull/1381#discussion_r31630497>:
>
> > +
> > +# Return a random value for the computer player input.
> > +fun computer_player : String do
> > +	var tab = ["rock", "scissors", "paper"]
> > +	return tab.rand
> > +end
> > +
> > +# Game loop.
> > +var x = ""
> > +var test = false
> > +loop
> > +	while not test do
> > +		print "Choose your weapon|exit"
> > +		x = gets
> > +		if x == "exit" then break
> > +		if x == "paper" or x == "rock" or x == "scissors" then test = true
>
> I do not recommend using ["paper", "rock", "scissors"].has(x) since a new
> Array will be allocated each time. Here this is not an issue but since it
> is in examples, it is not something I want to encourage as a good way to
> do things.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/privat/nit/pull/1381/files#r31630497>.
>